data-es6.js

```diff
@@ -6,6 +6,24 @@ var temp = {};
 var flag = "flagged";
 var fallthrough = "needs-polyfill-or-native";
 
+var typescript = {
+    compiler: { 
+        val: true, 
+        note_id: "typescript-compiler-or-native", 
+        note_html: "This feature is supported through a compiler error, or when the <code>--target ES6</code> flag is set and a native ES6 host is used." 
```

Comment:
"supported through a compiler error"...? We don't need that level of detail in these descriptions - it's assumed that for any compiler platform, a test that requires an error thrown via `eval()` should throw it when receiving such a line at compile-time.